src/components/Tabs/Scroller.tsx

```diff
@@ -0,0 +1,236 @@
+}
+
+const Scroller = ({
+  children,
+  activeChildIndex = 0,
+  center = false,
+  hideArrows = false,
+  arrowSize = 28,
+  breakoutPadding = 0
+}: ScrollerType) => {
+  const scrollRef = useRef<HTMLDivElement>()
+  const [{ left, right }, setArrows] = useState({ left: false, right: false })
+  const [scrollerOverflow, setOverflow] = useState(false)
+  const [colorScheme] = useColorContext()
+
+  useEffect(() => {
+    const scroller = scrollRef.current
+    if (!scroller) return
+    const target = Array.from(scroller?.children)[activeChildIndex + 1] // + 1 for pad element
+    scrollIntoView(target, {
+      time: 400,
+      align: {
+        left: 0,
+        leftOffset: breakoutPadding,
+        ...getTop()
+      }
+    })
+  }, [activeChildIndex])
+
+  useEffect(() => {
+    const scroller = scrollRef.current
+
+    const measure = () => {
+      let left = false
+      let right = false
+      if (scroller.scrollLeft > 0) {
+        left = true
+      }
+      if (scroller.scrollLeft + scroller.clientWidth < scroller.scrollWidth) {
+        right = true
+      }
+      setArrows(current => {
+        if (current.left !== left || current.right !== right) {
+          return { left, right }
+        }
+        return current
+      })
+      setOverflow(scroller.clientWidth < scroller.scrollWidth)
+    }
+    scroller.addEventListener('scroll', measure)
+    window.addEventListener('resize', measure)
+    measure()
+    return () => {
+      scroller.removeEventListener('scroll', measure)
+      window.removeEventListener('resize', measure)
+    }
+  }, [])
+
+  const getTop = () => {
+    const scroller = scrollRef.current
+    return {
+      top: 0,
+      topOffset: scroller.getBoundingClientRect().top
+    }
+  }
+
+  const handleArrowClick = direction => () => {
+    const scroller = scrollRef.current
+    const clientWidth = scroller.clientWidth
+    const target = Array.from(scroller.children).find(element => {
+      const { left, width } = element.getBoundingClientRect()
+      if (direction === 'left') {
+        return left + clientWidth >= 0
+      }
+      return left + width > clientWidth
+    })
+    // scroll all the way at the end
+    const newRightEdge =
+      scroller.scrollLeft + target.getBoundingClientRect().left + clientWidth
+    const leftOffset =
+      newRightEdge >= scroller.scrollWidth ? 0 : breakoutPadding
+
+    scrollIntoView(target, {
+      time: 400,
+      align: {
+        left: 0,
+        leftOffset,
+        ...getTop()
+      }
+    })
+  }
+
+  const shouldCenter = center && !scrollerOverflow
+
+  return (
+    <div {...styles.container} role='group'>
+      <div
+        ref={scrollRef}
+        {...styles.scroller}
+        style={{
+          justifyContent: shouldCenter ? 'center' : 'flex-start'
+        }}
+      >
+        <div
+          style={{
+            flex: shouldCenter
+              ? 1
+              : `0 0 ${scrollerOverflow ? breakoutPadding : 0}px`
```

Comment:
why would the padding only be rendered when overflowing? this make the spacing inconsistent between desktop and mobile right now
```suggestion
              : `0 0 ${breakoutPadding}px`
```